packages/ethereum-erc20-provider/lib/EthereumErc20Provider.js

```diff
@@ -76,6 +90,36 @@ export default class EthereumErc20Provider extends Provider {
       .filter(balance => !balance.isNaN())
       .reduce((acc, balance) => acc.plus(balance), BigNumber(0))
   }
+
+  async getMempoolBalance (addresses) {
+    // function balanceOf(address tokenOwner)
+    const functionSignature = '0x70a08231'
+    addresses = addresses
+      .map(address => String(address))
+    const addrs = addresses.map(ensure0x)
+    const balances = (await Promise.all(addrs.map(address => this.getMethod('jsonrpc')('eth_call', { data: functionSignature + padHexStart(removeAddress0x(address), 64), to: ensure0x(this._contractAddress) }, 'pending'))))
+      .map(balance => parseInt(balance, 16))
+      .reduce((acc, balance) => acc + balance, 0)
+    return isNaN(balances) ? 0 : balances
+  }
+
+  async getErc20Name () {
```

Comment:
Not really sure why we need this sort of functionality within the CAL. They cannot be replicated on any other chain therefore I think whichever the standard way of doing this in ethereum world would be a better approach than implementing all of this into the CAL.